airbyte-integrations/connectors/source-s3/v4_main.py

```diff
@@ -0,0 +1,16 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import sys
+
+from airbyte_cdk.entrypoint import AirbyteEntrypoint, launch
+from airbyte_cdk.sources.file_based.file_based_source import FileBasedSource
+from source_s3.v4 import Config, SourceS3StreamReader
+
+if __name__ == "__main__":
+    args = sys.argv[1:]
+    catalog_path = AirbyteEntrypoint.extract_catalog(args)
+    source = FileBasedSource(SourceS3StreamReader(), Config, catalog_path)
```

Comment:
Would it make sense to stand up the method stubs for the `S3FileSource` in this PR to help if the next set of PRs easier so there aren't conflicts modifying that central class?

Reply:
Could do, but I'm not anticipating that we'll need new methods on `FileBasedSource` so wasn't planning to create a separate `S3FileSource` (most of the customization occurs in the objects that are passed into `FileBasedSource`). Does that make sense to you or does it feel like there should be a dedicated `S3FileSource`?

Reply:
So I was sketching out the work for the S3 legacy config transformer to the new config. A lot of the config validation against the spec is done automatically in the entrypoint. So the potentially easiest way to transform legacy manifests before we perform the validation is to have an S3FileSource override `BaseConnector.read_config()`.

I'm still toying with ideas about how to do it best, but so far that was the simplest way to avoid having to manage multiple specs and versions of the config. But if you don't think your work will require it, then we won't conflict anyway

Reply:
Yeah what I'm doing now won't require it, so I think I'll let you make that change.